src/sparkle/application/__init__.py

```diff
@@ -1,68 +1,132 @@
 import abc
+import os
+from typing import Any, Dict, List
 from pyspark.sql import SparkSession, DataFrame
 from pyspark.storagelevel import StorageLevel
-from sparkle.config import Config
+from pyspark import SparkConf, SparkContext
+from pyspark.sql import SparkSession
+from pyspark.context import SparkContext
+from sparkle.config import Config, ExecutionEnvironment
 from sparkle.writer import Writer
 from sparkle.reader import Reader
+from sparkle.utils.logger import logger
 
-PROCESS_TIME_COLUMN = "process_time"
 
+_SPARK_EXTENSIONS = [
+    "org.apache.iceberg.spark.extensions.IcebergSparkSessionExtensions",
+]
+_SPARK_PACKAGES = [
+    "org.apache.iceberg:iceberg-spark-runtime-3.3_2.12:1.3.1",
+    "org.apache.spark:spark-sql-kafka-0-10_2.12:3.3.0",
+    "org.apache.spark:spark-avro_2.12:3.3.0",
+]
```

Comment:
You are defining protected property here, which means the end user should not have access to change this. However, you setting versions which can be deprecated based on a platform. I suggest to provide these as a default options but make it possible for the end user to change it.

Reply:
@farbodahm this needs to be passed using Spark's CLI arguments. Otherwise we need to lock Spark's version.